ui/cap-react/src/antd/schemas/Schemas.js

```diff
@@ -0,0 +1,263 @@
+  const [schema, setSchema] = useState();
+  const [field, setField] = useState();
+  const [selection, setSelection] = useState(FULL_SCHEMA);
+  const [options, setOptions] = useState();
+  const [showModal, setShowModal] = useState();
+  const [jsonError, setJsonError] = useState();
+  const [error, setError] = useState();
+
+  useEffect(() => {
+    axios
+      .get(`/api/jsonschemas/${schema_name}/${schema_version || ""}`)
+      .then(res => {
+        setOriginalSchema(res.data);
+        setSchema(res.data);
+        setField(res.data);
+        let opts = Object.entries(res.data)
+          .filter(
+            e =>
+              typeof e[1] === "object" && e[1] !== null && !Array.isArray(e[1])
+          )
+          .map(e => ({
+            label: e[0],
+            value: e[0],
+          }));
+        opts.unshift({ label: FULL_SCHEMA, value: "all" });
+        setOptions(opts);
+      })
+      .catch(() => {
+        setError(true);
+      });
+  }, []);
+
+  const handleSelectChange = (value, option) => {
+    if (value === "all") {
+      setField(schema);
+    } else {
+      setField(schema[value]);
+    }
+    setSelection(option.label);
+  };
+
+  const handleEdit = value => {
+    if (!value.error) {
+      const jsObject = value.jsObject;
+      if (selection === FULL_SCHEMA) {
+        setSchema(jsObject);
+      } else {
+        setSchema({ ...schema, [selection]: jsObject });
+      }
+      setField(jsObject);
+    }
+    setJsonError(value.error);
+  };
+
+  const handleSave = () => {
+    let schemaToSave = schema;
+    if (selection != FULL_SCHEMA) {
+      schemaToSave = { ...originalSchema, [selection]: schema[selection] };
+    }
+    axios
+      .put(`/api/jsonschemas/${schema_name}/${schema.version}`, schemaToSave)
+      .then(() => {
+        setOriginalSchema(schema);
+        notification.success({
+          message: "Schema updated",
+          description: "Changes successfully applied",
+        });
+      })
+      .catch(() =>
+        notification.error({
+          message: "Schema not updated",
+          description: "Error while saving, please try again",
+        })
+      );
+  };
+
+  const handleRevert = () => {
+    if (selection === FULL_SCHEMA) {
+      setField(originalSchema);
+      setSchema(originalSchema);
+    } else {
+      setField(originalSchema[selection]);
+      setSchema({ ...schema, [selection]: originalSchema[selection] });
+    }
+  };
+
+  return (
+    <DocumentTitle title="Schemas">
+      {error ? (
+        <ErrorScreen />
+      ) : (
+        <React.Fragment>
+          <Modal
+            visible={showModal}
+            onCancel={() => setShowModal(false)}
+            title={`${selection} diff`}
+            width={800}
+            footer={null}
+            bodyStyle={{
+              overflowX: "scroll",
+              overflowY: "auto",
+              maxHeight: "calc(100vh - 200px)",
+            }}
```

Comment:
Do we have separate css files? can move it there.

Reply:
I personally don't like extracting styling in antd to css/less files unless it's a big block or some more intricate styling. I think when it's just a few css properties leaving them in the js file is more practical. It's how it's also preferred in MUI (but with the `sx` prop instead of `styles`). It also seems to be the approach that has been followed with the existing antd code. What is you opinion @pamfilos?